state-chain/pallets/cf-elections/src/electoral_systems/monotonic_median.rs

```diff
@@ -114,3 +114,127 @@ impl<
+		pub static HOOK_HAS_BEEN_CALLED: std::cell::Cell<bool> = const { std::cell::Cell::new(false) };
+	}
+
+	const INIT_UNSYNCHRONISED_STATE: u64 = 22;
+
+	#[test]
+	fn check_consensus_correctly_calculates_median_when_all_authorities_vote() {
+		let mut electoral_system = MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+			INIT_UNSYNCHRONISED_STATE,
+			(),
+			(),
+		);
+		let mut votes = vec![((), 1), ((), 5), ((), 3), ((), 2), ((), 8), ((), 6), ((), 9)];
+		let number_of_authorities = votes.len() as u32;
+
+		use rand::{seq::SliceRandom, thread_rng};
+
+		votes.shuffle(&mut thread_rng());
+
+		let consensus = electoral_system
+			.new_election((), (), ())
+			.unwrap()
+			.check_consensus(None, votes, number_of_authorities)
+			.unwrap();
+
+		assert_eq!(consensus, Some(3));
+	}
+
+	#[test]
+	fn check_consensus_correctly_calculates_median_when_exactly_super_majority_authorities_vote() {
+		let mut electoral_system = MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+			INIT_UNSYNCHRONISED_STATE,
+			(),
+			(),
+		);
+		let mut votes = vec![((), 1), ((), 5), ((), 3), ((), 2), ((), 8), ((), 6), ((), 9)];
+		let number_of_authorities = votes.len() as u32;
+
+		use rand::{seq::SliceRandom, thread_rng};
+
+		votes.shuffle(&mut thread_rng());
+
+		let consensus = electoral_system
+			.new_election((), (), ())
+			.unwrap()
+			.check_consensus(None, votes, number_of_authorities + (number_of_authorities / 2))
+			.unwrap();
+
+		assert_eq!(consensus, Some(5));
+	}
+
+	#[test]
+	fn to_few_votes_consensus_not_possible() {
+		let mut electoral_system = MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+			INIT_UNSYNCHRONISED_STATE,
+			(),
+			(),
+		);
+		let votes = vec![((), 1), ((), 5), ((), 3), ((), 2), ((), 8), ((), 6)];
+		let number_of_authorities = votes.len() as u32;
+
+		let consensus = electoral_system
+			.new_election((), (), ())
+			.unwrap()
+			.check_consensus(None, votes, number_of_authorities * 2)
+			.unwrap();
+
+		assert_eq!(consensus, None);
+	}
+
+	#[test]
+	fn no_votes_consensus_not_possible() {
+		assert_eq!(
+			MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+				INIT_UNSYNCHRONISED_STATE,
+				(),
+				()
+			)
+			.new_election((), (), ())
+			.unwrap()
+			.check_consensus(None, vec![], 10)
+			.unwrap(),
+			None
+		);
+	}
+
+	#[test]
+	fn finalize_election() {
+		const INIT_UNSYNCHRONISED_STATE: u64 = 1;
+		const NEXT_UNSYNCHRONISED_STATE: u64 = 2;
+		let mut electoral_system = MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+			INIT_UNSYNCHRONISED_STATE,
+			(),
+			(),
+		);
+		let mut election = electoral_system.new_election((), (), ()).unwrap();
+		election.set_consensus_status(ConsensusStatus::Changed {
+			previous: INIT_UNSYNCHRONISED_STATE,
+			new: NEXT_UNSYNCHRONISED_STATE,
+		});
+		electoral_system.finalize_elections(&()).unwrap();
+		// Hock has been called
+		assert!(HOOK_HAS_BEEN_CALLED.with(|hook_called| hook_called.get()));
```

Comment:
Let's encapsulate the raw static variable accesses behind a method in the Mock definition.